core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -144,7 +162,17 @@ public static <T extends StructLike> PositionDeleteIndex toPositionIndex(
             deletes ->
                 CloseableIterable.transform(
                     locationFilter.filter(deletes), row -> (Long) POSITION_ACCESSOR.get(row)));
-    return toPositionIndex(CloseableIterable.concat(positions));
+    return toPositionIndex(positions);
+  }
+
+  public static PositionDeleteIndex toPositionIndex(List<CloseableIterable<Long>> positions) {
+    PositionDeleteIndex positionDeleteIndex = new BitmapPositionDeleteIndex();
```

Comment:
I don't think this should duplicate the logic of the other method. It should still concatenate into a `CloseableIterable`. But if there is more than one `CloseableIterable` it should use a `ParallelIterable` to do it.

Reply:
Refactored this in the recent PR.